exercises/poker/canonical-data.json

```diff
@@ -0,0 +1,317 @@
+      ]
+    },
+    {
+      "description": "multiple hands with the same high cards, tie compares next highest ranked, down to last card",
+      "property": "bestHands",
+      "input": [
+        "3S 5H 6S 8D 7H",
+        "2S 5D 6D 8C 7S"
+      ],
+      "expected": [
+        "3S 5H 6S 8D 7H"
+      ]
+    },
+    {
+      "description": "one pair beats high card",
+      "property": "bestHands",
+      "input": [
+        "4S 5H 6C 8D KH",
+        "2S 4H 6S 4D JH"
+      ],
+      "expected": [
+        "2S 4H 6S 4D JH"
+      ]
+    },
+    {
+      "description": "highest pair wins",
+      "property": "bestHands",
+      "input": [
+        "4S 2H 6S 2D JH",
+        "2S 4H 6C 4D JD"
+      ],
+      "expected": [
+        "2S 4H 6C 4D JD"
+      ]
+    },
+    {
+      "description": "two pairs beats one pair",
+      "property": "bestHands",
+      "input": [
+        "2S 8H 6S 8D JH",
+        "4S 5H 4C 8C 5C"
+      ],
+      "expected": [
+        "4S 5H 4C 8C 5C"
+      ]
+    },
+    {
+      "description": "both hands have two pairs, highest ranked pair wins",
+      "property": "bestHands",
+      "input": [
+        "2S 8H 2D 8D 3H",
+        "4S 5H 4C 8S 5D"
+      ],
+      "expected": [
+        "2S 8H 2D 8D 3H"
+      ]
+    },
+    {
+      "description": "both hands have two pairs, with the same highest ranked pair, tie goes to low pair",
+      "property": "bestHands",
+      "input": [
+        "2S QS 2C QD JH",
+        "JD QH JS 8D QC"
+      ],
+      "expected": [
+        "JD QH JS 8D QC"
+      ]
+    },
+    {
+      "description": "both hands have two identically ranked pairs, tie goes to remaining card (kicker)",
+      "property": "bestHands",
+      "input": [
+        "JD QH JS 8D QC",
+        "JS QS JC 2D QD"
+      ],
+      "expected": [
+        "JD QH JS 8D QC"
+      ]
+    },
+    {
+      "description": "three of a kind beats two pair",
+      "property": "bestHands",
+      "input": [
+        "2S 8H 2H 8D JH",
+        "4S 5H 4C 8S 4H"
+      ],
+      "expected": [
+        "4S 5H 4C 8S 4H"
+      ]
+    },
+    {
+      "description": "both hands have three of a kind, tie goes to highest ranked triplet",
+      "property": "bestHands",
+      "input": [
+        "2S 2H 2C 8D JH",
+        "4S AH AS 8C AD"
+      ],
+      "expected": [
+        "4S AH AS 8C AD"
+      ]
+    },
+    {
+      "description": "with multiple decks, two players can have same three of a kind, ties go to highest remaining cards",
+      "property": "bestHands",
+      "input": [
+        "4S AH AS 7C AD",
+        "4S AH AS 8C AD"
+      ],
+      "expected": [
+        "4S AH AS 8C AD"
+      ]
+    },
+    {
+      "description": "a straight beats three of a kind",
+      "property": "bestHands",
+      "input": [
+        "4S 5H 4C 8D 4H",
+        "3S 4D 2S 6D 5C"
+      ],
+      "expected": [
+        "3S 4D 2S 6D 5C"
+      ]
+    },
+    {
+      "description":"aces can end a straight (10 J Q K A)",
```

Comment:
> > potential case to add: two hands that each have the same pair, to exercise comparison of the other cards
>
> I believe that case is already taken care of by this test: "both hands have two identically ranked pairs, tie goes to remaining card (kicker)".

Interesting. I hadn't thought of it that way, since that was a two-pair hand rather than a one-pair hand, but I can see it working out.

OK, just that extra space

... which you just put in, ok.